### ISSUE TITLE
Incorrect escaping of markdown note content

### ISSUE THREAD

**Issue by michal-organek** [BoostIO/Boostnote#1360]
**Read my last comment at https://github.com/BoostIO/Boostnote/issues/1360#issuecomment-355236359**

Probably issue is related to unlinking one of storages (I removed one storage via "Storages"), because it is only unusual  action I did.

I had created 150 separated note pages, and after I unlink one of storages it cannot see them anymore, and cannot see new notes I'm adding.
After I add new note, next close and open Boostnote again, new note disappears.
It always shows the same 8 notes (I have no idea why those once).
However when I check storage files at my disk, I can see all of notes.
Boostnote can see storages and folders structure.

Can you help me?

Boostnote version: 0.8.19
```bash
$ lsb_release -a
No LSB modules are available.
Distributor ID:	LinuxMint
Description:	Linux Mint 18.3 Sylvia
Release:	18.3
Codename:	sylvia
```

**Comment by f4il** [comment]
It is what I can see at DebugConsole after start Boostnote - not sure it is related to issue.
![selection_373](https://user-images.githubusercontent.com/1150261/34533228-a1313316-f0b9-11e7-865a-a8afb027b6f8.png)

**Comment by f4il** [comment]
I have found a reason of issue.

**It seems that there is incorrect escaping on markdown content.**

I removed notes that I found in above screenshot and Boostnote started working as expected (`22b1221de3285fe98a5c.cson`).

In removed note I found line:
```
 $6$lcLoLcJt$2d2rF51GwOWMJSAEgexa1rVavJIkdBY9EK6T3sJrRcsfzasgEBhIIRYWyWmXTRyGpxECizcoMcYq8Tj5rJv0u0\'''''
```

Originally pasted into note line was:
```
$6$lcLoLcJt$2d2rF51GwOWMJSAEgexa1rVavJIkdBY9EK6T3sJrRcsfzasgEBhIIRYWyWmXTRyGpxECizcoMcYq8Tj5rJv0u0'''''
```

After I removed `\''''` error doesn't occur anymore.
So I guess that application interprets `'''` at the end of line, as the end of note markdown content.

**Comment by Flexo013** [comment]
Sorry that this has gone unnoticed for such a long time.

Is this issue still present in the latest version of Boostnote?

**Comment by Flexo013** [comment]
I'm closing this issue do to prolonged inactivity.

If you still are experiencing this problem then feel free to reopen this issue or create a new issue with all the relevant information.

**Comment by RichardWPerrott** [comment]
Indeed, HTML links can be broken by incorrect Markdown quoting for pasted page content

# Current behavior
`abstract [Optional](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html "class in java.util")<[CookieHandler](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/CookieHandler.html "class in java.net")>`

# Expected behavior
Only quote text around link, never quote around link target, () part, and maybe pointless/incompatible around the label, [] part, too.
`abstract`[HttpClient.Redirect](https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/HttpClient.Redirect.html "enum in java.net.http")

# Steps to reproduce
Open (https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/HttpClient.html) in a browser, select all, copy the content, then paste into a new Markdown Note in Boostnote.

# Environment
- Version : 0.12.1
- OS Version and name : Microsoft Window 10 Professional

**Comment by Flexo013** [comment]
Closed as per #2913
